test/src/worker/browser-env/runner/test-runner/index.ts

```diff
@@ -357,20 +375,119 @@ describe("worker/browser-env/runner/test-runner", () => {
 
         describe(`"${WorkerEventNames.finalize}" event`, () => {
             it(`should emit after completing the test`, async () => {
-                sandbox.stub(NodejsEnvRunner.prototype, "run").resolves();
+                sandbox.stub(NodejsEnvRunner.prototype, "finishRun").resolves();
 
                 const socket = mkSocket_();
                 socketClientStub.returns(socket);
 
                 await runWithEmitBrowserInit(socket);
 
                 assert.callOrder(
-                    NodejsEnvRunner.prototype.run as SinonStub,
+                    NodejsEnvRunner.prototype.finishRun as SinonStub,
                     (socket.emit as SinonStub).withArgs(WorkerEventNames.finalize),
                 );
             });
         });
 
+        describe(`"${BrowserEventNames.reconnect}" event`, () => {
+            let clock: SinonFakeTimers;
+
+            beforeEach(() => {
+                clock = sinon.useFakeTimers({ toFake: ["setInterval", "setTimeout", "clearInterval"] });
+            });
+
+            afterEach(() => clock.restore());
+
+            describe("should throw error if", () => {
+                it("browser not inited after reconnect", async () => {
+                    const socket = mkSocket_() as BrowserViteSocket;
+                    socketClientStub.returns(socket);
+
+                    const promise = run_();
+                    await clock.tickAsync(1);
```

Comment:
used in order to give event loop ability to run source code (subscribe on `initialize` event)